urllib3/response.py

```diff
@@ -432,11 +432,17 @@ def read_chunked(self, amt=None, decode_content=None):
             'content-encoding' header.
         """
         self._init_decoder()
-        # FIXME: Rewrite this method and make it a class with
-        #        a better structured logic.
+        # FIXME: Rewrite this method and make it a class with a better structured logic.
         if not self.chunked:
             raise ResponseNotChunked("Response is not chunked. "
                 "Header 'transfer-encoding: chunked' is missing.")
+
+        if self._original_response and self._original_response._method.upper() == 'HEAD':
+            # Don't bother reading the body of a HEAD request.
+            # FIXME: Can we do this somehow without accessing private httplib _method?
```

Comment:
Wasn't there an issue about attaching request information to a response somehow? Or was that about tracking redirect history somehow? Was there one for each? I forget.